x-pack/plugins/security_solution/server/lib/detection_engine/signals/executors/eql.ts

```diff
@@ -0,0 +1,129 @@
+import { buildEqlSearchRequest } from '../../../../../common/detection_engine/get_query_filter';
+import { hasLargeValueItem } from '../../../../../common/detection_engine/utils';
+import { ExceptionListItemSchema } from '../../../../../common/shared_imports';
+import { isOutdated } from '../../migrations/helpers';
+import { getIndexVersion } from '../../routes/index/get_index_version';
+import { MIN_EQL_RULE_INDEX_VERSION } from '../../routes/index/get_signals_template';
+import { RefreshTypes } from '../../types';
+import { buildSignalFromEvent, buildSignalGroupFromSequence } from '../build_bulk_body';
+import { getInputIndex } from '../get_input_output_index';
+import { RuleStatusService } from '../rule_status_service';
+import { bulkInsertSignals, filterDuplicateSignals } from '../single_bulk_create';
+import {
+  EqlRuleAttributes,
+  EqlSignalSearchResponse,
+  SearchAfterAndBulkCreateReturnType,
+  WrappedSignalHit,
+} from '../types';
+import { createSearchAfterReturnType, makeFloatString, wrapSignal } from '../utils';
+
+export const eqlExecutor = async ({
+  rule,
+  exceptionItems,
+  ruleStatusService,
+  services,
+  version,
+  searchAfterSize,
+  logger,
+  refresh,
+}: {
+  rule: SavedObject<EqlRuleAttributes>;
+  exceptionItems: ExceptionListItemSchema[];
+  ruleStatusService: RuleStatusService;
+  services: AlertServices<AlertInstanceState, AlertInstanceContext, 'default'>;
+  version: string;
+  searchAfterSize: number;
+  logger: Logger;
+  refresh: RefreshTypes;
+}): Promise<SearchAfterAndBulkCreateReturnType> => {
+  const result = createSearchAfterReturnType();
+  const ruleParams = rule.attributes.params;
+  if (hasLargeValueItem(exceptionItems)) {
+    await ruleStatusService.partialFailure(
+      'Exceptions that use "is in list" or "is not in list" operators are not applied to EQL rules'
+    );
+    result.warning = true;
+  }
+  try {
+    const signalIndexVersion = await getIndexVersion(
+      services.scopedClusterClient.asCurrentUser,
+      ruleParams.outputIndex
+    );
+    if (isOutdated({ current: signalIndexVersion, target: MIN_EQL_RULE_INDEX_VERSION })) {
+      throw new Error(
+        `EQL based rules require an update to version ${MIN_EQL_RULE_INDEX_VERSION} of the detection alerts index mapping`
+      );
+    }
+  } catch (err) {
+    if (err.statusCode === 403) {
+      throw new Error(
+        `EQL based rules require the user that created it to have the view_index_metadata, read, and write permissions for index: ${ruleParams.outputIndex}`
+      );
+    } else {
+      throw err;
+    }
+  }
+  const inputIndex = await getInputIndex(services, version, ruleParams.index);
+  const request = buildEqlSearchRequest(
+    ruleParams.query,
+    inputIndex,
+    ruleParams.from,
+    ruleParams.to,
+    searchAfterSize,
+    ruleParams.timestampOverride,
+    exceptionItems,
+    ruleParams.eventCategoryOverride
+  );
+  const eqlSignalSearchStart = performance.now();
+  // TODO: fix this later
```

Comment:
ah yeah this is referring to the use of `transport.request` still, which was necessary with the legacy ES client because the EQL apis weren't in it - but the EQL api is in the new client, so we can switch over.